2-completed-app/service/tableClient.js

```diff
@@ -0,0 +1,104 @@
+const { TableClient } = require("@azure/data-tables");
+const env = require("../configure/env")
+const serviceClient = TableClient.fromConnectionString(
+  env.connectionString,
+  env.tableName
+);
+
+const getData = async function(){
+    try{
+        const resArray = [];
+        const res = serviceClient.listEntities();
+    
+        for await(const entity of res){
+            resArray.push(entity)
+        }
+        return {code:0,data:resArray};
+    }catch(Error){
+        return {code:1,error:Error};
+    }
+ 
+}
+const filterData = async function(option){
+    const resArray = [];
+    const filters = [];
+    if(option.partitionKey!==null){
```

Comment:
I think you should check for undefined here. if you get the following object this condition will be `true`

maybe you want to do this
```javascript
if(option.partitionKey)
```

which is true if `partitionKey` is not `null`, `undefined`, `""`, `false` or `0`

Reply:
This probably applies to all the checks in this file where you are checking for not null